File: src/Hedera.Hashgraph.Tests/HookExtensionPointTest.cs
// SPDX-License-Identifier: Apache-2.0
using Org.Junit.Jupiter.Api;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;

namespace Com.Hedera.Hashgraph.Sdk
{
    class HookExtensionPointTest
    {
        virtual void RoundTripAllEnumValues()
        {
            foreach (HookExtensionPoint value in HookExtensionPoint.Values())
            {
                var proto = value.GetProtoValue();
                AssertNotNull(proto);
                var restored = HookExtensionPoint.FromProtobuf(proto);
                AssertEquals(value, restored, "Round-trip mismatch for " + value);
            }
        }

        virtual void ProtoValuesAreStable()
        {
            foreach (HookExtensionPoint value in HookExtensionPoint.Values())
            {
                var proto = value.GetProtoValue();

                // basic sanity: ordinal-like mapping should not be negative
                AssertTrue(proto.GetNumber() >= 0);
                AssertTrue(value.ToString().Length > 0);
            }
        }
    }
}

File: src/test/java/com/hedera/hashgraph/sdk/HookExtensionPointTest.java
// SPDX-License-Identifier: Apache-2.0
package com.hedera.hashgraph.sdk;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class HookExtensionPointTest {

    @Test
    void roundTripAllEnumValues() {
        for (HookExtensionPoint value : HookExtensionPoint.values()) {
            var proto = value.getProtoValue();
            assertNotNull(proto);
            var restored = HookExtensionPoint.fromProtobuf(proto);
            assertEquals(value, restored, "Round-trip mismatch for " + value);
        }
    }

    @Test
    void protoValuesAreStable() {
        for (HookExtensionPoint value : HookExtensionPoint.values()) {
            var proto = value.getProtoValue();
            // basic sanity: ordinal-like mapping should not be negative
            assertTrue(proto.getNumber() >= 0);
            assertTrue(value.toString().length() > 0);
        }
    }
}
